src/app/components/event-page/event-page.component.ts

```diff
@@ -0,0 +1,132 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, takeUntil } from 'rxjs';
+import { ICategory } from 'src/app/models/category';
+import { Event } from 'src/app/models/event';
+import { EventLocation } from 'src/app/models/event-location';
+import { CategoryService } from 'src/app/services/category.service';
+import { DialogService } from 'src/app/services/dialog.service';
+import { EndpointsService } from 'src/app/services/endpoints.service';
+import { EventService } from 'src/app/services/event.service';
+@Component({
+  selector: 'app-event-page',
+  templateUrl: './event-page.component.html',
+  styleUrls: ['./event-page.component.scss'],
+})
+export class EventPageComponent implements OnInit {
+  eventId: number;
+  isHostMode: true;
+  event: Event = {
```

Comment:
Please prefix your interfaces with an "I". There is also a built in Event interface that represents an event which takes place in the DOM.